### PR TITLE
Use sha256 for rpm signing to allow installation on fips-enabed systems

### LINKED ISSUE #4267: Unable to install grafana-agent rpm on FIPS-enabled OS
### What's wrong?

RPM complains about Digest mismatch, while installing `grafana-agent` on FIPS-installed OS:
```
# fips-mode-setup --check
FIPS mode is enabled.

# rpm -i grafana-agent-0.34.1-1.amd64.rpm
error: unpacking of archive failed on file /etc/grafana-agent.yaml;649b0319: cpio: Digest mismatch
error: grafana-agent-0.34.1-1.x86_64: install failed
```

This happens because MD5 is set as the digest singing algorithm, and FIPS prohibits usage of MD5 anywhere:
```
# rpm -qp --qf "%{filedigestalgo}\n"  grafana-agent-0.34.1-1.amd64.rpm
(none)
```
`(none)` here means MD5 , as per https://bugzilla.redhat.com/show_bug.cgi?id=1659053

This is different grafana (not the agent) rpm, which as of https://github.com/grafana/grafana/pull/59510 is now signed with sha256:
```
# rpm -qp --qf "%{filedigestalgo}\n"  grafana-7.5.15-4.el8.x86_64.rpm
8
```

### Steps to reproduce

1. Download grafana-agent rpm from the upstream repo to a FIPS-enabled OS
2. Attempt to install it with `rpm -i grafana-agent-0.34.1-1.amd64.rpm`

### System information

RHEL 8.8

### Software version

grafana-agent-0.34.1-1

### Configuration

```text
FIPS enabled
```

### Logs

_No response_

### PR DESCRIPTION
**Pull request by kfriedrich123**
#### PR Description

Sets gpg digest signing algo to sha256 to allow rpm installation on FIPS enabled OS'es

#### Which issue(s) this PR fixes

Fixes #4267 

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Comment by rfratto** [comment]
Thanks for the PR! I have to hand this off to the team at Grafana which introduced the signing to make sure this is OK, since the agent maintainers don't know too much about how this works.

**Comment by rfratto** [comment]
It sounds like this PR is safe to merge, but where are you getting the packages from? If you're getting them from rpm.grafana.com, it sounds like that team might have to look into changing the signing there to support FIPS-enabled systems, since the packages hosted on rpm.grafana.com are re-signed.

**Review by rfratto — APPROVED** [review]
Thanks!

**Comment by kfriedrich123** [comment]
Hey @rfratto , thank you for a quick merge and release! 

We get the packages from rpm.grafana.com. Do you know if the re-signing process is also open-source? I'm happy to test/contribute to speed up the process, are we're currently blocked on rolling out grafana-agent to our FIPS-enabled hosts.

**Pull request merged**
